test/unit/polyfills/test-resize-observer.js

```diff
@@ -0,0 +1,330 @@
+    element1 = win.document.createElement('div');
+    element2 = win.document.createElement('div');
+  });
+
+  it('should allow default constructor', () => {
+    expect(() => new ResizeObserverStub(callback)).to.not.throw();
+  });
+
+  describe('observe/unobserve/disconnect', () => {
+    it('should queue up elements when observed, but only once', () => {
+      const ro = new ResizeObserverStub(callback);
+      ro.observe(element1);
+      expect(ro.elements_).to.deep.equal([element1]);
+      ro.observe(element1);
+      expect(ro.elements_).to.deep.equal([element1]);
+      ro.observe(element2);
+      expect(ro.elements_).to.deep.equal([element1, element2]);
+    });
+
+    it('should dequeue elements when observed', () => {
+      const ro = new ResizeObserverStub(callback);
+      ro.observe(element1);
+      ro.observe(element2);
+      expect(ro.elements_).to.deep.equal([element1, element2]);
+      ro.unobserve(element1);
+      expect(ro.elements_).to.deep.equal([element2]);
+      ro.unobserve(element2);
+      expect(ro.elements_).to.deep.equal([]);
+      ro.unobserve(element1);
+      ro.unobserve(element2);
+      expect(ro.elements_).to.deep.equal([]);
+    });
+
+    it('should dequeue elements when disconnected', () => {
+      const ro = new ResizeObserverStub(callback);
+      ro.observe(element1);
+      ro.observe(element2);
+      expect(ro.elements_).to.deep.equal([element1, element2]);
+      ro.disconnect();
+      expect(ro.elements_).to.deep.equal([]);
+      ro.disconnect();
+      expect(ro.elements_).to.deep.equal([]);
+    });
+  });
+
+  describe('upgrade', () => {
+    beforeEach(() => {
+      env.sandbox.stub(NativeResizeObserver.prototype, 'observe');
+      env.sandbox.stub(NativeResizeObserver.prototype, 'unobserve');
+      env.sandbox.stub(NativeResizeObserver.prototype, 'disconnect');
+    });
+
+    function upgrade(io) {
+      io.upgrade_(NativeResizeObserver);
+      return io.inst_;
+    }
+
+    it('should forward callback', () => {
+      const ro = new ResizeObserverStub(callback);
+      const native = upgrade(ro);
+      expect(native.callback).to.equal(callback);
+    });
+
+    it('should not re-queue if nothing is currently observed', () => {
+      const ro = new ResizeObserverStub(callback);
+      ro.observe(element1);
+      ro.unobserve(element1);
+      const native = upgrade(ro);
+      expect(native.observe).to.not.be.called;
+      expect(ro.elements_).to.be.null;
+    });
+
+    it('should re-queue previously observed elements', () => {
+      const ro = new ResizeObserverStub(callback);
+      ro.observe(element1);
+      ro.observe(element2);
+      const native = upgrade(ro);
+      expect(native.observe).to.be.calledTwice;
+      expect(native.observe).to.be.calledWith(element1);
+      expect(native.observe).to.be.calledWith(element2);
+      expect(ro.elements_).to.be.null;
+    });
+
+    it('should observe new elements only on native', () => {
```

Comment:
For this and the following tests, should you be checking that `ro.elements_` doesn't have any methods called on it?

Reply:
The `ro.elements_` is set to `null`, so that if accidentally it does get used - it simply fails right away.

Reply:
Ah! Good point